bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,134 @@
+        strings_found = self.find_with_start_end(
+                                    filename, file, strings, escape=True)
+        # multiline Comments
+        comments_found = self.find_with_start_end(
+                filename, file, dict(lang_dict["multiline_comment_delimiters"]))
+        # single-line Comments
+        comments_found += self.find_singleline_comments(
+                           filename, file, list(lang_dict["comment_delimiter"]))
+
+        matches_found = strings_found + comments_found
+        # Remove Nested
+        matches_found = tuple(filter(
+                              lambda arg: not starts_within_range(matches_found,
+                                                                  arg),
+                              matches_found))
+        # Yield results
+        strings_found = tuple(filter(lambda arg: arg in matches_found,
+                                     strings_found))
+        yield HiddenResult(self, strings_found)
+        comments_found = tuple(filter(lambda arg: arg in matches_found,
+                                      comments_found))
+        yield HiddenResult(self, comments_found)
+
+    @staticmethod
+    def find_with_start_end(filename, file, annot, escape=False):
+        """
+        Gives all positions of annotations which have a start and end.
+
+        :param filename: Name of the file on which bear is running.
+        :param file:     Contents of the file in the form of tuple of lines.
+        :param annot:    A dict containing start of annotation as key and end of
+                         annotation as value.
+        :param escape:   Variable to check wether to ignore escaped annotations.
+
+        :return:         A tuple of SourceRanges giving the range of annotation.
+        """
+        text = ''.join(file)
+        if escape:
+            search_func = unescaped_search_in_between
+        else:
+            search_func = search_in_between
+        found_pos = ()
+        for annot_type in annot:
+            found_pos += tuple(search_func(annot_type, annot[annot_type], text))
+        if found_pos:
+            found_pos = tuple(SourceRange.from_absolute_position(
+                                filename,
+                                AbsolutePosition(file, position.begin.range[0]),
+                                AbsolutePosition(file, position.end.range[1]-1))
+                              for position in found_pos)
+        return found_pos
+
+    @staticmethod
+    def find_singleline_comments(filename, file, comments):
+        """
+        Finds all single-line comments.
+
+        :param filename: Name of the file on which bear is running.
+        :param file:     Contents of the file in the form of tuple of lines.
+        :param comments: A list contatining different types of
+                         single-line comments.
+        :return:         A tuplle of SourceRange objects with start as start of
```

Comment:
you sure you can't formulate this a little better? feels insecure with ``start as start`` & ``end as end``